### PR TITLE
Map file management

### PR DESCRIPTION
**Pull request by adscib**
Here's my proposal for how we should manage the map files. We would bundle the CARLA maps with the `torchdrivesim` package itself, and provide a mechanism to include additional paths for maps files using the `TDS_RESOURCE_PATH` environment variable. All the files are stored in a safe and human-readable format, mostly JSON and OSM. There's also a function to download and save all the map information through the IAI API. Traffic light control will be added in #42.

I included the right-handed Town03 map as an example, but ultimately I'm planning to replace it with all the left-handed CARLA maps before merging. You can run `python -m examples.simulate` to see that everything runs out of the box, and it will produce `simulation.gif` in the main folder.

Let me know what you think and if everyone's happy I can finish this, include the map files we want, and add a couple of tests.

### CODE REVIEW

**Review by zhangk1551 — APPROVED** [review]
It's much cleaner in this way. Thank you!

File: torchdrivesim/map.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    resource_path = torchdrivesim._resource_path
+    for root in resource_path:
+        map_path = os.path.join(root, map_name)
+        if os.path.exists(map_path):
+            break
+    else:
+        map_path = None
+    if map_path is None:
+        return None
```
Comment (Ruishenl) [inline torchdrivesim/map.py:131]:
```suggestion
        return None
```

File: examples/simulate.py [hunk 2]
```diff
@@ -56,11 +56,37 @@ def simulate(cfg: SimulationConfig):
...
+    if traffic_control_poses['traffic-light']:
+        traffic_controls['traffic_light'] = TrafficLightControl(
+            torch.stack(traffic_control_poses['traffic-light']).unsqueeze(0)
+        )
+    if traffic_control_poses['stop-sign']:
+        traffic_controls['stop_sign'] = StopSignControl(
+            torch.stack(traffic_control_poses['stop-sign']).unsqueeze(0)
+        )
+    if traffic_control_poses['yield-sign']:
+        traffic_controls['yield_sign'] = TrafficLightControl(
```
Comment (Ruishenl) [inline examples/simulate.py]: This should be `YieldControl`  instead ?

File: examples/simulate.py [hunk 3]
```diff
@@ -56,11 +56,37 @@ def simulate(cfg: SimulationConfig):
...
+            torch.stack(traffic_control_poses['yield_sign']).unsqueeze(0)
```
Comment (Ruishenl) [inline examples/simulate.py]: This should be `yield-sign`

**Review by Ruishenl — APPROVED** [review]

**Comment by adscib** [comment]
I think this is ready to merge, but let me know if you spotted anything bad in my recent commits.

**Review by Ruishenl — APPROVED** [review]

**Comment by zhangk1551** [comment]
@adscib 
Hi Adam, I just noticed the carla maps here only contain Town01~Town04 and Town10HD, could we include Town07 as well? It's used in the `torchdriveenv` related data, including training and validation scenario cases.

The [rl-env labeling tool](https://internal.inverted.ai/rl-boxes) supports `"Town01", "Town02",  "Town03", "Town04",  "Town06", "Town07", "Town10HD"`. For `torchdriveenv`, `Town04` and `Town06 are excluded, because they are too large and will cause issues in the API server side.

**Pull request merged**